src/main/java/org/jabref/gui/groups/GroupTreeViewModel.java

```diff
@@ -260,6 +260,50 @@ boolean onlyMinorChanges(AbstractGroup oldGroup, AbstractGroup newGroup) {
         return true;
     }
 
+    /**
+     * Opens "Rename Group Dialog" and change name of group
+     */
+    public void renameGroup(GroupNodeViewModel oldGroup) {
+        currentDatabase.ifPresent(database -> {
+            AbstractGroup oldGroupDef = oldGroup.getGroupNode().getGroup();
+            String oldGroupName = oldGroupDef.getName();
+
+            Optional<AbstractGroup> newGroup = dialogService.showCustomDialogAndWait(
+                    new RenameGroupView(database,
+                            oldGroup.getGroupNode().getGroup())
+            );
+
+            newGroup.ifPresent(group -> {
+
+                String newGroupName = group.getName();
+
+                if (newGroupName.trim().isEmpty()) {
+                    return;
+                }
+
+                if (newGroupName.contains(",")) {
+                    return;
+                }
```

Comment:
Doesn't JabRef have a check for new group names?

What about the dialog to add new groups?

If the check is also in there, maybe, the method can be extracted?